File: TimeSlicer/TimePiece.cs
using System;

namespace TimeSlicer
{
    public class TimePiece
    {
        public TimePiece(string time)
        {
            Hour = Convert.ToInt32(time.Split(Convert.ToChar(":"))[0]);
            MilitaryHour = Convert.ToInt32(time.Split(Convert.ToChar(":"))[0]) + 12;
            Minute = Convert.ToInt32(time.Split(Convert.ToChar(":"))[1]);
        }

        public string Error { get; set; }
        public int Hour { get; set; }
        public int MilitaryHour { get; set; }
        public int Minute { get; set; }
    }
}

File: TimeSliceJavaScript/TimeSlicer/TimePiece.js
/**
 * User: Adron Hall
 * Date: 8/18/11
 * Time: 3:31 PM
 * This library provides functionality to parse a time string value.
 */

TimePiece = function(time) {

    TimePiece.Hour = function () {
        return parseInt(time.toString().split(":")[0]);
    }

    TimePiece.Minute = function () {
        var timeString = time.toString();
        timeString = timeString.split(":")[1];
        timeString = timeString.substr(0, 2);
        return parseInt(timeString);
    }

    TimePiece.Meridian = function() {
        var timeString = time.toString();
        timeString = timeString.split(":")[1];
        timeString = timeString.substr(timeString.length - 2, 2).toUpperCase();
        return timeString;
    }

    TimePiece.MilitaryHour = function () {
        if (TimePiece.Meridian() == "AM")
            return TimePiece.Hour();
        return parseInt(TimePiece.Hour()) + 12;
    };

    return time;
};
